lldb/packages/Python/lldbsuite/test/tools/lldb-dap/dap_server.py

```diff
@@ -1187,15 +1187,17 @@ def request_locations(self, locationReference):
         }
         return self.send_recv(command_dict)
 
-    def request_testGetTargetBreakpoints(self):
+    def request_testGetTargetBreakpoints(self, only_resolved=False):
```

Comment:
I'm not sure we need to extend this, we could track the current state of breakpoints in `DebugCommunication` by updating `_handle_recv_packet` to check for breakpoint events and by updating `request_setBreakpoints` (and other kinds of breakpoints functions) to update the state when they get a response from the server.

We could have a helper for getting the current state of a breakpoint by id and move `wait_for_breakpoints_to_resolve` into to `DebugCommunication`.

Reply:
Good idea to move this to `DebugCommunication` but I think we still need to have this request because the breakpoints that resolve immediately don't generate the `breakpoint` async event

Reply:
They should be marked `verified` in the response to the `setBreakpoint` request, so if we update our state in that call when we get the response then it should be consistent.

Reply:
Nice catch :100: